bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorStore.java

```diff
@@ -170,14 +174,23 @@ private void flushErrorReport(File errorFile) {
             Logger.warn("Could not send previously saved error(s)"
                 + " to Bugsnag, will try again later", exception);
         } catch (Exception exception) {
-            if (delegate != null) {
-                Error minimalError = generateErrorFromFilename(errorFile.getName());
+            deleteStoredFiles(Collections.singleton(errorFile));
+        }
+    }
 
-                if (minimalError != null) {
-                    delegate.onErrorReadFailure(minimalError);
-                }
+    @Nullable
```

Comment:
This should not be nullable. In the event that the filename metadata cannot be parsed fully, it should make a default error selecting from `Exception`:`error`:`unhandled` for whichever parts are missing.